### PR TITLE
Build numpy from src in integration test

### PR DESCRIPTION
**Pull request by ryanking13**
Build numpy using `pyodide build` command in the integration test.

### CODE REVIEW

**Review by agriyakhetarpal — COMMENTED** [review]
Thanks, @ryanking13! NumPy builds with Meson, so this is nice to see. Could we also add one more Scientific Python project, say, one that relies on a different build backend such as `scikit-build-core`? We are not constrained by the number of CI jobs, and there would be scope for parallelising them.

File: integration_tests/src/numpy.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -e
+
+URL="https://files.pythonhosted.org/packages/65/6e/09db70a523a96d25e115e71cc56a6f9031e7b8cd166c1ac8438307c14058/numpy-1.26.4.tar.gz"
```
[the remark this answers is not in the document]
Comment (ryanking13) [reply integration_tests/src/numpy.sh:5]: Sounds good to me. I didn't know that sdist has such format BTW.

File: integration_tests/src/numpy.sh
[the remark this answers is not in the document]
Comment (hoodmane) [reply integration_tests/src/numpy.sh:1]: We are never establishing windows support. People on Windows can use windows subsystem linux.

[the remark this answers is not in the document]
Comment (agriyakhetarpal) [reply integration_tests/src/numpy.sh:1]: I see, no worries! There is a chance things might work for anyone trying it out with minimal changes (except for the shell scripts, I guess), so if someone wants, they can do those changes themselves.

**Comment by ryanking13** [comment]
I updated the target to Numpy v2.0.2. I'll go ahead and merge this PR. Lets add more packages in the follow-up PRs.

**Pull request merged**